airlock/business_logic.py

```diff
@@ -54,6 +56,62 @@ class FileReviewStatus(Enum):
     REJECTED = "REJECTED"
 
 
+class AuditEventType(Enum):
+    """Audit log events."""
+
+    # file access
+    WORKSPACE_FILE_VIEW = "WORKSPACE_FILE_VIEW"
+    REQUEST_FILE_VIEW = "REQUEST_FILE_VIEW"
+    REQUEST_FILE_DOWNLOAD = "REQUEST_FILE_DOWNLOAD"
+
+    # request status
+    REQUEST_CREATE = "REQUEST_CREATE"
+    REQUEST_SUBMIT = "REQUEST_SUBMIT"
+    REQUEST_WITHDRAW = "REQUEST_WITHDRAW"
+    REQUEST_APPROVE = "REQUEST_APPROVE"
+    REQUEST_REJECT = "REQUEST_REJECT"
+    REQUEST_RELEASE = "REQUEST_RELEASE"
+
+    # request file status
+    REQUEST_FILE_ADD = "REQUEST_FILE_ADD"
+    REQUEST_FILE_WITHDRAW = "REQUEST_FILE_WITHDRAW"
+    REQUEST_FILE_APPROVE = "REQUEST_FILE_APPROVE"
+    REQUEST_FILE_REJECT = "REQUEST_FILE_REJECT"
+
+
+@dataclass
+class AuditEvent:
+    type: AuditEventType
+    user: str
+    workspace: str | None = None
+    request: str | None = None
+    path: str | None = None
```

Comment:
why `str` and not `UrlPath`?

Reply:
Because all of the others are simple python types, and its stored as a string in the db.

We could change it, but it just moves the conversion around into the DAL.

Reply:
Have moved to UrlPath